### PR TITLE
Slight refactors in preparation for #10480

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

Code operating on store objects (including creating them) should, in general, use `ContentAddressMethod` rather than `FileIngestionMethod`.

# Context

#10480

See also dfc876531f269950a4e183a4f77a813c421d7d64 which included some similar refactors.

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: src/libexpr/primops.cc [hunk 1]
```diff
@@ -2395,7 +2394,9 @@ static void prim_path(EvalState & state, const PosIdx pos, Value * * args, Value
         else if (n == "filter")
             state.forceFunction(*(filterFun = attr.value), attr.pos, "while evaluating the `filter` parameter passed to builtins.path");
         else if (n == "recursive")
-            method = FileIngestionMethod { state.forceBool(*attr.value, attr.pos, "while evaluating the `recursive` attribute passed to builtins.path") };
```
Comment (Ericson2314) [inline src/libexpr/primops.cc:2398]: Casting a bool is bad for relying on internal representation details too much, so I made it more explicit.

**Review by roberth — CHANGES_REQUESTED** [review]
I can't easily infer why this is an improvement, based on the documentation of the `ContentAddressMethod` and `FileIngestionMethod` types.

Specifically, it'd be good to describe how they relate, and perhaps add some guidance on which to pick, if we even need both.

**Comment by Ericson2314** [comment]
Hehe very fair. https://github.com/NixOS/nix/pull/10479 and a follow up for store object content addressing are supposed to *finally, thoroughly* answer that question. OK leaving this open until they are done!

**Comment by Ericson2314** [comment]
https://github.com/NixOS/nix/pull/10722 is the (barely started draft) of the follow-up I mentioned.

**Comment by Ericson2314** [comment]
OK I hope that https://github.com/NixOS/nix/pull/10722 is far enough along that we can make a decision on this :)

**Review by roberth — APPROVED** [review]

**Pull request merged**